### ISSUE TITLE
bug: LazyGit not working with latest version and neovim-ayu theme

### ISSUE THREAD

**Issue by liamwh** [LazyVim/LazyVim#2829]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.9.5

### Operating system/version

Arch Linux

### Describe the bug

```
E5108: Error executing lua: ...cal/share/nvim/lazy/LazyVim/lua/lazyvim/util/lazygit.lua:42: standard file is closed
stack traceback:
	[C]: in function 'write'
	...cal/share/nvim/lazy/LazyVim/lua/lazyvim/util/lazygit.lua:42: in function 'set_ansi_color'
	...cal/share/nvim/lazy/LazyVim/lua/lazyvim/util/lazygit.lua:47: in function 'update_theme'
	...cal/share/nvim/lazy/LazyVim/lua/lazyvim/util/lazygit.lua:32: in function 'lazygit'
	...l/share/nvim/lazy/LazyVim/lua/lazyvim/config/keymaps.lua:128: in function <...l/share/nvim/lazy/LazyVim/lua/lazyvim/config/keymaps.lua:128>
```

### Steps To Reproduce

Use LazyGit with space + gg when in a Git repo with the latest LazyVim version. (commit: 20efdd44da9473b23aba503de2865fb0b1868fd4)

I also have this in my plugins dir:

```lua
return {
    {
        "Shatur/neovim-ayu",
    },
    {
        "LazyVim/LazyVim",
        opts = {
            colorscheme = "ayu",
        },
    },
}
```

### Expected Behavior

LazyGit to open successfully.

**Comment by dpetka2001** [comment]
Can't reproduce. This is what I get with your configuration 
```lua
return {
    {
        "Shatur/neovim-ayu",
    },
    {
        "LazyVim/LazyVim",
        opts = {
            colorscheme = "ayu",
        },
    },
}
```

![2024-03-26_16-04](https://github.com/LazyVim/LazyVim/assets/12776461/0db71f02-19af-45de-b50a-8b1bfa2e221d)

**Comment by folke** [comment]
That seems to fail becasue your stdout stream of Neovim is closed.

No idea why that might be.

Just updated this to pcall it instead. Should no longer error (but you'll be missing one color override)

**Comment by akthe-at** [comment]
Have installed the #2831 release and there still might be an issue going on with the LazyGit change. I can't see the error message when I use <leader>gg because the window disappears too quickly but if I open a term split and try to type lazygit here, in the same directory mind you, it says "2024/03/26 09:51:02 CreateFile WARNING: at <ScriptBlock>: The filename, directory name, or volume label syntax is incorrect."

**Comment by dpetka2001** [comment]
You should have a file created in the equivalent for Windows `~/.cache/nvim/lazygit-theme.yml`. Does it get created? Maybe it has to do with the different Windows path separator and it should be accounted for in the `M.theme_path`?

**Comment by akthe-at** [comment]
> You should have a file created in the equivalent for Windows `~/.cache/nvim/lazygit-theme.yml`. Does it get created? Maybe it has to do with the different Windows path separator and it should be accounted for in the `M.theme_path`?

This directory did exist but was empty when I checked it just now.

**Comment by dpetka2001** [comment]
What does `:echo stdpath("data")` show for you in Windows?

**Comment by akthe-at** [comment]
@dpetka2001  C:\Users\MyUserName\AppData\Local\nvim-data

**Comment by dpetka2001** [comment]
In that folder can you open the file `lazy/LazyVim/lua/lazyvim/util/lazygit.lua` and change line 10 to `M.theme_path = vim.fn.stdpath("cache") .. package.config:sub(1, 1) .. "lazygit-theme.yml"` and check if it then works?

**Comment by akthe-at** [comment]
copy and pasting my current line just in case I typoed something but with this change, no dice. 

`M.theme_path = vim.fn.stdpath("cache") .. package.config:sub(1, 1) .. "/lazygit-theme.yml"
`

**Comment by dpetka2001** [comment]
Remove the forward slash from string `lazygit-theme.yml`. It should just be `M.theme_path = vim.fn.stdpath("cache") .. package.config:sub(1, 1) .. "lazygit-theme.yml"`

**Comment by folke** [comment]
right, that might be it.

Your issue reported you used Arch Linux, but so this is on Windows?

Did @dpetka2001's edit fix it?

**Comment by dpetka2001** [comment]
@folke This is a different user than OP, hence the different OS (Windows).

**Comment by akthe-at** [comment]
Yes, different user, yes Windows 10, sorry for the confusion thought it would be better than spamming a new issue of the same type. did not fix, going to revert to yesterday's lazyvim version quick to make sure I didn't do something else.

**Comment by dpetka2001** [comment]
It would be useful if you could provide the error that happens then. This was just an assumption from my part since you use Windows.

**Comment by folke** [comment]
After trying to invoke lazygit, what does `:=vim.env.LG_CONFIG_FILE` show?

**Comment by akthe-at** [comment]
Same Error as initially posted,

> 2024/03/26 09:51:02 CreateFile WARNING: at : The filename, directory name, or volume label syntax is incorrect.

New breakthrough in debugging. 
Lazygit does not work with this new option set
`LazyVim.terminal.setup("pwsh")`

if I comment that out, restore to the latest version of lazyvim without all the above suggested changes and instead use the following...Lazygit now works
```
vim.o.shell = "C:/Users/ARK010/scoop/apps/git/current/bin/bash.exe"
local gitbash_options = {
  shell = vim.fn.executable("C:/Users/ARK010/scoop/apps/git/current/bin/bash.exe") == 1
    and "C:/Users/ARK010/scoop/apps/git/current/bin/bash.exe",
  shellcmdflag = "-c",
  shellredir = ">%s 2>&1",
  shellpipe = "2>&1| tee",
  shellquote = "",
  shellxquote = "",
}

for option, value in pairs(gitbash_options) do
  vim.opt[option] = value
end
```

**Comment by akthe-at** [comment]
> After trying to invoke lazygit, what does `:=vim.env.LG_CONFIG_FILE` show?
I am sorry for how ugly this output looks, there appears to be some escaping issues.

```
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[33;1mWARNING: at <ScriptBlock>, C:\Users\ARK010\Documents\Powershell\Microsoft.PowerShell_profile.ps1: line 116
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE at <ScriptBlock>, <No file>: line 1
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1mSet-PSReadLineOption: ^[[0mC:\Users\ARK010\Documents\Powershell\Microsoft.PowerShell_profile.ps1:116^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1mLine |^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1m^[[36;1m 116 | ^[[0m ^[[36;1mSet-PSReadLineOption -PredictionViewStyle ListView^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1m^[[36;1m^[[0m^[[36;1m^[[0m^[[36;1m     | ^[[31;1m ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1m^[[36;1m^[[0m^[[36;1m^[[0m^[[36;1m^[[31;1m^[[31;1m^[[36;1m     | ^[[31;1mThe handle is invalid.^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[33;1mWARNING: at <ScriptBlock>, C:\Users\ARK010\Documents\PowerShell\Microsoft.PowerShell_profile.ps1: line 116
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE at <ScriptBlock>, <No file>: line 1
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1mSet-PSReadLineOption: ^[[0mC:\Users\ARK010\Documents\PowerShell\Microsoft.PowerShell_profile.ps1:116^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1mLine |^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1m^[[36;1m 116 | ^[[0m ^[[36;1mSet-PSReadLineOption -PredictionViewStyle ListView^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1m^[[36;1m^[[0m^[[36;1m^[[0m^[[36;1m     | ^[[31;1m ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE ^[[31;1m^[[0m^[[36;1m^[[36;1m^[[0m^[[36;1m^[[0m^[[36;1m^[[31;1m^[[31;1m^[[36;1m     | ^[[31;1mThe handle is invalid.^[[0m
10:58:40 AM msg_show   vim.env.LG_CONFIG_FILE C:\Users\ARK010\AppData\Local\lazygit/config.yml,C:\Users\ARK010\AppData\Local\Temp\nvim/lazygit-theme.yml


```

**Comment by akthe-at** [comment]
Here is that line of code from the Powershell Profile that was referenced as line 116 `Set-PSReadLineOption -PredictionViewStyle ListView
`

When I commented this line out, Lazygit started working with `LazyVim.terminal.setup("pwsh")` being set again.

**Comment by akthe-at** [comment]
Instead of just commenting out 
`Set-PSReadLineOption -PredictionViewStyle ListView `

I changed it to 
`Set-PSReadLineOption -PredictionViewStyle InlineView` for now and Lazygit works inside of neovim. I am fine with this fix. I know it can be difficult to make things work across all these platforms.

**Comment by dpetka2001** [comment]
Can this be passed as `shellcmdflag` in `LazyVim.terminal.setup()`? Unfortunately, I don't know much about Windows. Also, is my suggestion still needed or it works without it? You should undo the change you did in the `data` folder of LazyVim, so you don't have any problems with updating.

**Comment by folke** [comment]
I just logged in on my Windows machine, also using pwsh and everything works as expected for me

**Comment by dpetka2001** [comment]
Also in [Microsoft docs](https://learn.microsoft.com/en-us/powershell/module/psreadline/set-psreadlineoption?view=powershell-7.4) it says
> Sets the style for the display of the predictive text. The default is InlineView.
    InlineView - the style as existing today, similar as in fish shell and zsh. (default)
    ListView - suggestions are rendered in a drop down list, and users can select using UpArrow and DownArrow.

Is it safe to assume this is the default option and you changed it yourself?

**Comment by akthe-at** [comment]
Correct, I changed the default InlineView to ListView in my Powershell profile, since everything works just fine with InlineView I am good with the current self-fix. I am also okay with using git-bash as well! I have reverted all of my changes that you suggested @dpetka2001 because they were not needed to fix the issue. At least this information is here in case another windows user comes nocking. Thank you both for the help!
